adds federated service architecture and explains working (#23)
hypertrace/hypertrace-federated-service: README.md

[![CircleCI](https://circleci.com/gh/hypertrace/hypertrace-federated-service.svg?style=svg)](https://circleci.com/gh/hypertrace/hypertrace-federated-service)

> Hypertrace federated service combines [Hypertrace-UI](https://github.com/hypertrace/hypertrace-ui), [Hypertrace-GraphQL](https://github.com/hypertrace/hypertrace-graphql), [Gateway service](https://github.com/hypertrace/gateway-service), [Attribute service](https://github.com/hypertrace/attribute-service), [Query service](https://github.com/hypertrace/query-service), [Entity service](https://github.com/hypertrace/entity-service) into single service. This is used in hypertrace standalone deployment to make the deployment compact and use less resource.

Reviewer: ```suggestion
Hypertrace Federated Service combines [Hypertrace-UI](https://github.com/hypertrace/hypertrace-ui), [Hypertrace-GraphQL](https://github.com/hypertrace/hypertrace-graphql), [Gateway service](https://github.com/hypertrace/gateway-service), [Attribute service](https://github.com/hypertrace/attribute-service), [Query service](https://github.com/hypertrace/query-service), [Entity service](https://github.com/hypertrace/entity-service) into single service. This is used in hypertrace standalone deployment to make the deployment compact and use less resource.
```

Reviewer 2: yeah I think we shouldn't bother with changing query-service etc to "Query Service" as it makes more work when renaming and less concrete to container names.

> ## How Federated service works?

Reviewer: ```suggestion
## How Hypertrace Federated Service works?
```

| ![space-1.jpg](https://hypertrace-docs.s3.amazonaws.com/federated-service.png) | 
|:--:| 
| *Hypertrace Query Architecture* |

> In Hypertrace Federated service, 

Reviewer: ```suggestion
In Hypertrace Federated Service, 
```

- GraphQL service is being used by UI and attribute-service fetches all relevant attributes to the scope of what is being shown.

> - gateway-service provides single access point which routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by downstream services. 

Reviewer 3: ```suggestion
- gateway-service provides single access point which routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by upstream services. 
```

> - gateway-service provides single access point which routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by upstream services. 

Reviewer 4: ```suggestion
- The gateway-service provides a single access point that routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by upstream services. 
```
